### PR TITLE
Add notes on packaging for distributions.

### LINKED ISSUE #8292: Provide some documentation for package maintainers
We should document some packaging best practices and configuration nobs to make it easier to package deal.II for various *nix distributions.

In reference to #8290, #8291

### PR DESCRIPTION
**Pull request by drwells**
Fixes #8292.

### CODE REVIEW

File: doc/developers/packaging.html [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  <body>
+
+    <h1>Packaging <acronym>deal.II</acronym> for a Unix-like distribution</h1>
+
+    <h2>Overview</h2>
+    <p>
+      This page describes how to write packaging scripts
+      for <acronym>deal.II</acronym>. <acronym>deal.II</acronym> is written in
+      C++11 and is known to work with fairly recent GCC, Clang, and Intel
```
Comment (bangerth) [inline doc/developers/packaging.html]: Actually, we go pretty far back I think. I'd just drop the "fairly recent".

File: doc/developers/packaging.html [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      architecture for dependencies: all dependencies must be provided at
+      compile-time.
+
+      The only required dependency of <acronym>deal.II</acronym> is boost.
+    </p>
+
+
+    <h2>Examples</h2>
+    <acronym>deal.II</acronym> is already available in many Linux distributions:
+    for example packaging scripts see, e.g.,
```
Comment (bangerth) [inline doc/developers/packaging.html]:
```suggestion
    for examples of packaging scripts see, e.g.,
```

File: doc/developers/packaging.html [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      <li>
+        <acronym>deal.II</acronym> defaults to installing several
+        configuration files
+        (<tt>LICENSE.md</tt>, <tt>README.md</tt>, <tt>detailed.log</tt>,
+        and <tt>summary.log</tt>) in the installation prefix. We recommend
+        deleting these files. <tt>LICENSE.md</tt> should be installed in the
+        correct directory for software licenses.
+      </li>
+      <li>
+        By default, all examples are compiled in both debug and release mode and
```
Comment (bangerth) [inline doc/developers/packaging.html]:
```suggestion
        By default, all tutorial programs are compiled in both debug and release mode and
```

File: doc/developers/packaging.html [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      </li>
+      <li>
+        The flag <tt>DEAL_II_WITH_64BIT_INDICES</tt> changes an integral type
+        used for indices in the library. This value should be made consistent
+        with the way PETSc is configured.
+      </li>
+      <li>
+        <acronym>deal.II</acronym> includes copies of boost, muParser, TBB, and
+        UMFPACK. These bundled packages should be disabled
+        via <tt>-DDEAL_II_ALLOW_BUNDLED=OFF<tt>.
```
Comment (bangerth) [inline doc/developers/packaging.html]:
```suggestion
        via <tt>-DDEAL_II_ALLOW_BUNDLED=OFF<tt> and these dependencies should instead
        be resolved by ensuring that whenever deal.II is installed, the packages above are also
        installed.
```

File: doc/developers/packaging.html [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      </li>
+      <li>
+        Underlinkage with GSL may cause problems, depending on how GSL is set up.
```
Comment (bangerth) [inline doc/developers/packaging.html]: This may require some more explanation :-)

**Comment by drwells** [comment]
Thanks for the comments. I fixed all of them and added a changelog entry.

**Comment by bangerth** [comment]
@kronbichler -- were you waiting for something here after you approved the patch?

**Comment by kronbichler** [comment]
Not really, except if someone else had something to say. It is long overdue, so I will merge.

**Pull request merged**